ament_cpplint/ament_cpplint/main.py

```diff
@@ -199,27 +205,59 @@ def custom_error(filename, linenum, category, confidence, message):
     return 1 if _cpplint_state.error_count else 0
 
 
-def get_file_groups(paths, extensions):
+def is_file_or_directory_to_exclude(path_to_file, excludes):
+    if not os.path.isdir(path_to_file):
+        path, file_name = os.path.split(path_to_file)
+    else:
+        path = path_to_file
+        file_name = ''
+    path_to_check = pathlib.Path(path)
+
+    def check_path_parents(parents, exclude):
+        for p in parents:
+            if p.match(exclude):
+                return True
+        return False
+
+    for excl in excludes:
+        if (
+            excl == file_name or
+            path_to_check.match(excl) or
+            check_path_parents(path_to_check.parents, excl)
+        ):
+            return True
+    return False
+
+
+def get_file_groups(paths, extensions, excludes):
     # dict mapping root path to files
     groups = {}
     for path in paths:
         if os.path.isdir(path):
             for dirpath, dirnames, filenames in os.walk(path):
+                if is_file_or_directory_to_exclude(dirpath, excludes):
+                    continue
+
                 if 'AMENT_IGNORE' in dirnames + filenames:
                     dirnames[:] = []
                     continue
                 # ignore folder starting with . or _
                 dirnames[:] = [d for d in dirnames if d[0] not in ['.', '_']]
                 dirnames.sort()
+                for dirname in dirnames:
+                    if is_file_or_directory_to_exclude(os.path.join(dirpath, dirname), excludes):
+                        continue
```

Comment:
This loop doesn't do anything. It needs to remove excluded directories from `dirnames` instead.

Also there shouldn't be a need for a recursive check like `is_file_or_directory_to_exclude`.

Reply:
You just requested another review on this PR but I don't see any update after this comment?

Reply:
I pushed in the wrong remote  https://github.com/ament/ament_lint/pull/238/commits/201486a59d695a9b304187d7c739584ee8640e99